kimchi/src/proof.rs

```diff
@@ -2,126 +2,225 @@
 
 use crate::circuits::wires::{COLUMNS, PERMUTS};
 use ark_ec::AffineCurve;
-use ark_ff::{FftField, Zero};
+use ark_ff::{FftField, Field, Zero};
 use ark_poly::univariate::DensePolynomial;
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
 use array_init::array_init;
 use commitment_dlog::{commitment::PolyComm, evaluation_proof::OpeningProof};
 use o1_utils::{types::fields::*, ExtendedDensePolynomial};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_with::serde_as;
 
 //~ spec:startcode
-#[derive(Clone)]
-pub struct LookupEvaluations<Field> {
+#[serde_as]
+#[derive(Clone, Deserialize, Serialize)]
+pub struct LookupEvaluations<F: CanonicalSerialize + CanonicalDeserialize> {
     /// sorted lookup table polynomial
-    pub sorted: Vec<Field>,
+    #[serde_as(as = "Vec<Vec<o1_utils::serialization::SerdeAs>>")]
+    pub sorted: Vec<Vec<F>>,
     /// lookup aggregation polynomial
-    pub aggreg: Field,
+    #[serde_as(as = "Vec<o1_utils::serialization::SerdeAs>")]
+    pub aggreg: Vec<F>,
     // TODO: May be possible to optimize this away?
     /// lookup table polynomial
-    pub table: Field,
-
+    #[serde_as(as = "Vec<o1_utils::serialization::SerdeAs>")]
+    pub table: Vec<F>,
     /// Optionally, a runtime table polynomial.
-    pub runtime: Option<Field>,
+    #[serde_as(as = "Option<Vec<o1_utils::serialization::SerdeAs>>")]
+    pub runtime: Option<Vec<F>>,
```

Comment:
Lookups are never chunked?

Reply:
I set this one as `Vec<F>` because in the previous version, `Field` could potentially be a vector. Nonetheless, I will check now which of them could be just single elements, both for this one and for other fields of the structs.